backend/de.metas.contracts/src/main/java/de/metas/contracts/FlatrateTermRequest/ModularFlatrateTermRequest.java

```diff
@@ -37,7 +39,9 @@
 public class ModularFlatrateTermRequest
 {
 	@NonNull BPartnerId bPartnerId;
-	@NonNull ProductId productId;
-	@Nullable YearId yearId;
+	@Nullable ProductId productId;
 	@NonNull SOTrx soTrx;
+	@Nullable YearId harvestingYearId;
```

Comment:
@metas-ts @metas-rc  is it ok to call those fields harvestingYearId and harvestingCalendarId? don't we have a much more generic names but still descriptive?

Reply:
In the domain for which we are building this functionality, the year is always a harvesting year, since we deal with harvested grains. But I can see this type of contract being reused in other domains that are not related with harvesting. Renamed them to yearId and calendarId.